### ISSUE TITLE
Option to prevent connecting to Godot editor

### ISSUE THREAD

**Issue by ithinkandicode** [godotengine/godot-vscode-plugin#434]
### Godot version

3.5-stable

### VS Code version

1.72.2

### Godot Tools VS Code extension version

1.3.1

### System information

Window 10

### Problem statement

This notices popups up if you aren't running Godot:

> Couldn't connect to the GDScript language server at 127.0.0.1:6008. Is the Godot editor running?

I only want to use the syntax highlighting feature of this extension, so is there a way to prevent it from trying to connect to the editor every time?

I'm aware of #356 but the fix was to run Godot, which isn't a viable fix for this situation

### Proposed solution

Add an option to prevent trying to connect to the editor

**Comment by DaelonSuzuka** [comment]
I like this idea. I'll probably add a "don't bother me again" button to the popup, and an option in the extension settings. 

Thanks for the suggestion.

**Comment by Calinou** [comment]
> I'll probably add a "don't bother me again" button to the popup

This sounds too easy to accidentally activate, especially if it applies globally and not on a per-folder basis.

**Comment by DaelonSuzuka** [comment]
> This sounds too easy to accidentally activate, especially if it applies globally and not on a per-folder basis.

That's definitely possible. I was planning to play around with it. Maybe it could be a "don't bother me again today", or "don't bother me again this session" without risking getting stuck. Also, thanks to oidro's great PR fixing all the settings names, I now know how to save state per-workspace, so maybe "don't bother me again today, for this workspace" could be sufficiently "safe".

**Comment by Calinou** [comment]
We should probably make it so the setting ignores connection errors, rather than disabling the connection entirely even if it could work.

**Comment by AlfishSoftware** [comment]
Agree. Also, I think changing the popup buttons is not strictly necessary, as settings can already be applied per-workspace if desired (you could just add a text like "This warning can be disabled in settings." to the popup).
So a setting like proposed should be enough:
```jsonc
/** Disable the popup warning when all connection attempts at startup fail. */
"godot_tools.warn_unable_to_connect": false, // default true
```
The setting should be considered only at startup, it should be ignored if you clicked Retry.

By the way, when trying to find workarounds for this using the existing settings, here's what happened:
```jsonc
"godot_tools.reconnect_attempts": 0, // tried to connect 11 times (10+1), instead of just once
"godot_tools.reconnect_attempts": -1, // immediate warning; presumably tried to connect once, instead of 0
"godot_tools.gdscript_lsp_server_port": -1, // stuck on "Initializing" forever; clicking it shows popup:
// "Connecting to the GDScript language server at 127.0.0.1:-1" so it seems there's no validation of port
```
